src/mscorlib/shared/System/IO/PathHelper.Windows.cs

```diff
@@ -131,10 +115,10 @@ private static string TryExpandShortFileName(ref StringBuffer outputBuffer, stri
             //
             // Note that we will never get \??\ here as GetFullPathName() does not recognize \??\ and will return it as C:\??\ (or whatever the current drive is).
 
-            int rootLength = PathInternal.GetRootLength(outputBuffer.AsSpan());
-            bool isDevice = PathInternal.IsDevice(outputBuffer.AsSpan());
+            int rootLength = PathInternal.GetRootLength(outputBuilder.AsSpan());
+            bool isDevice = PathInternal.IsDevice(outputBuilder.AsSpan());
 
-            StringBuffer inputBuffer = new StringBuffer(0);
+            ValueStringBuilder inputBuilder = new ValueStringBuilder();
```

Comment:
Is it not worth starting with stack space?

Reply:
I didn't want to add another .5k to the stack for the corner case. I can if you think I should.

Reply:
Maybe worth a comment.